public/pages/utils/anomalyResultUtils.ts

```diff
@@ -483,3 +523,274 @@ export const parsePureAnomalies = (
+  };
+
+  const allFeatures = get(
+    detector,
+    'featureAttributes',
+    [] as FeatureAttributes[]
+  );
+  allFeatures.forEach(feature => {
+    //@ts-ignore
+    const featureData = featuresData[feature.featureId];
+    const featureDataPoints = getFeatureDataPoints(
+      featureData,
+      interval,
+      dateRange
+    );
+
+    featureDataPointsForDetector = {
+      ...featureDataPointsForDetector,
+      [feature.featureName]: featureDataPoints,
+    };
+  });
+  return featureDataPointsForDetector;
+};
+
+export const getFeatureMissingSeverities = (featuresDataPoint: {
+  [key: string]: FeatureDataPoint[];
+}): Map<MISSING_FEATURE_DATA_SEVERITY, string[]> => {
+  const featureMissingSeverities = new Map();
+
+  for (const [featureName, featureDatePoints] of Object.entries(
+    featuresDataPoint
+  )) {
+    // all feature data points should have same length
+    let featuresWithMissingData = [] as string[];
+    if (featureDatePoints.length <= 1) {
+      // return empty map
+      return featureMissingSeverities;
+    }
+    if (
+      featureDatePoints.length === 2 &&
+      featureDatePoints[0].isMissing &&
+      featureDatePoints[1].isMissing
+    ) {
+      if (featureMissingSeverities.has(MISSING_FEATURE_DATA_SEVERITY.YELLOW)) {
+        featuresWithMissingData = featureMissingSeverities.get(
+          MISSING_FEATURE_DATA_SEVERITY.YELLOW
+        );
+      }
+      featuresWithMissingData.push(featureName);
+      featureMissingSeverities.set(
+        MISSING_FEATURE_DATA_SEVERITY.YELLOW,
+        featuresWithMissingData
+      );
+      continue;
+    }
+
+    const orderedFeatureDataPoints = orderBy(
+      featureDatePoints,
+      // sort by plot time in desc order
+      dataPoint => get(dataPoint, 'plotTime', 0),
+      SORT_DIRECTION.DESC
+    );
+    // feature has >= 3 data points
+    if (
+      orderedFeatureDataPoints.length >= 3 &&
+      orderedFeatureDataPoints[0].isMissing &&
+      orderedFeatureDataPoints[1].isMissing
+    ) {
+      // at least latest 2 ones are missing
+      let currentSeverity = MISSING_FEATURE_DATA_SEVERITY.YELLOW;
+      if (orderedFeatureDataPoints[2].isMissing) {
+        // all the latest 3 ones are missing
+        currentSeverity = MISSING_FEATURE_DATA_SEVERITY.RED;
+      }
+      if (featureMissingSeverities.has(currentSeverity)) {
+        featuresWithMissingData = featureMissingSeverities.get(currentSeverity);
+      }
+      featuresWithMissingData.push(featureName);
+      featureMissingSeverities.set(currentSeverity, featuresWithMissingData);
+    }
+  }
+
+  return featureMissingSeverities;
+};
+
+export const getFeatureDataMissingMessageAndActionItem = (
+  featureMissingSev: MISSING_FEATURE_DATA_SEVERITY | undefined,
+  featuresWithMissingData: string[]
+) => {
+  switch (featureMissingSev) {
+    case MISSING_FEATURE_DATA_SEVERITY.YELLOW:
+      return {
+        message: `Recent data is missing for feature ${featuresWithMissingData.join(
```

Comment:
if > 1 feature with missing data, should probably say "Recent data is missing for features" rather than "feature"

Reply:
`feature` here is more like adjective, because feature name(s) is shown after `feature`. I will confirm this with tech writer.